### ISSUE TITLE
At-launch WCAG = digitoegankelijk = webrichtlijnen compliance

### ISSUE THREAD

**Issue by alexdekok** [minvws/nl-covid19-notification-app-website#60]
Voldoen om te testen of de site voldoet aan eisen van digitoegankelijkheid.nl
Toegankelijkheids statement en test kunnen na lancering

Bekijk de laatste stand van zaken: https://minvws.github.io/nl-covid19-notification-app-website/

## Definition of done
- [x] Powermapper om te checken of we voldoen aan WCAG-richtlijnen https://www.powermapper.com/  (A en AA niveau)
- [x] Review by Bram Duvigneau

## Stuff to check:
- [x] Toegankelijkheid v/d website algemeen
- [x] Toegankelijkheid info slider
- [x] Toegankelijkheid statements (vierkante blokken op home)
- [x] Toegankelijkheid faq (veelgestelde vragen)

**Comment by erikkroes** [comment]
Een korte blik op de gegeven link geeft mij de indruk dat er nog veel te doen valt. Er zijn zelfs verbeterpunten die met een automatische test al naar voren komen.
Nu lees ik je bericht nog wat grondiger en zie ik: 
> Toegankelijkheids statement en test kunnen na lancering
Wat is er veranderd sinds de eerste berichten waar gezegd werd dat toegankelijkheid centraal zou staan? Een product dat niet toegankelijk is is niet af, en dat moet je niet lanceren.

**Comment by mellewynia** [comment]
Thanks to your comment @erikkroes and the help of the community Fronteers. I have made several technical improvements which should improve the accessibility greatly. If you are interested, please feel free to review the following related commits:

- https://github.com/minvws/nl-covid19-notification-app-website/commit/aec00c42fd728d6a02e04bc278193e0285949de5
- https://github.com/minvws/nl-covid19-notification-app-website/commit/8459e7462dee02aa5caa9e6217418a67984cc7e8
- https://github.com/minvws/nl-covid19-notification-app-website/commit/f12615b4f80ea90cbdcbb05e9a351ed6ec9d880b

When it comes to powermapper: not all issues have been addressed. Please note: quite a few reported issues are not applicable. There are however a few (design) concerns as well, that should be addressed (cc: @HarrieKuipers ).

- In the carrousel (info-slider) there is text that is styled as underlined. The way to emphasise certain words should be changed, since this way of styling will confuse users thinking a word is a link. Suggestion: maybe make it bold or emphasise it with a background color. 
- A carroussel should have a pause button. I have added one, however there is not styling for it as of yet. 
- I do not believe the carroussel to be fully accessible as of yet. I am exploring how this could be improved.

**Comment by gberkouwer** [comment]
Helemaal eens met de opmerking van @erikkroes : "Toegankelijkheids statement en test kunnen na lancering". Het is een terechte verplichting om dit bij livegang goed te hebben, zowle het vloden aan als het statement. 'Toegankelijkheid by design en by default', niet achteraf.

**Comment by bramd** [comment]
We gaan snel live met deze website, zo kunnen we ook snel feedback van gebruikers verzamelen. We doen ons best de site bij livegang zo toegankelijk mogelijk te hebben.

## Toegankelijkheid

We hebben vanmiddag een hands-on sessie gedaan rondom toegankelijkheid, zie het als een smoketest. Daarin hebben we een hoop kunnen oplossen, maar het is natuurlijk niet een diepgaande audit. Suggesties zijn altijd welkom om de site verder te verbeteren en de openstaande issues worden later alsnog opgepakt.

Een korte samenvatting van de verbeteringen:

[-] Skiplink: goed
[-] Taalselectie: voldoende, hertesten als dropdown geïmplementeerd is
[-] Carrousel (info slider): voldoende, was onvoldoende
    [-] Afbeeldingen zijn decoratief + kijken naar background image
    [-] Slider knoppen: goed
    [-] Statements (popup): goed
    [-] FAQ: tab-acchtige interface: voldoende
    [-] Feedback button: goed
        [ ] Linktekst 'hier' vervangen
[ ] Pauzeer slider knop toegevoegd: ontwerp van knop in orde maken, was niet ontworpen
[ ] Stippen in orde maken, voldiet nu niet aan contrast WCAG AA
[ ] Footer: logo: Logo alleen samen ... kleurstelling onhandig voor slechtzienden, maar geen afkeuring op WCAG omdat het een logo betreft
[-] Tabel caption te licht van kleur: is nu goed, was een contrastprobleem

Opmerking: Vanuit toegankelijkheid/usability is een slider niet aan te bevelen. Dit zorgt altijd voor complexiteit, zowel wat techniek als bediening voor de gebruiker betreft.

## Usability

De eerste versie van de website zal morgen ook worden meegenomen in gebruikerstests voor de app.

**Comment by erikkroes** [comment]
> We doen ons best de site bij livegang zo toegankelijk mogelijk te hebben.

Ik hoop dat iedereen dat doet, maar ik kan niet meer denk dan: "Daar koop je niks voor". Mag deze software uberhaupt ontoegankelijk gelanceerd worden?

**Comment by mellewynia** [comment]
@EdoPlantinga, n.a.v. stand-up: deze issue In Progress gezet daar we nog wel wat restpunten hebben. Die pakken we gaanderwijs op.

**Comment by alexdekok** [comment]
@mellewynia  check jij dan svp ook even de mail van vanmorgen en de punten die open staan in ticket#45 en dan vooral doelend op de terugkoppeling op het document wat daar is toegevoegd: https://github.com/minvws/nl-covid19-notification-app-website/issues/45

**Comment by bramd** [comment]
> > We doen ons best de site bij livegang zo toegankelijk mogelijk te hebben.
> 
> Ik hoop dat iedereen dat doet, maar ik kan niet meer denk dan: "Daar koop je niks voor". Mag deze software uberhaupt ontoegankelijk gelanceerd worden?

Ik snap dat je dat denkt, zo zou ik waarschijnlijk ook reageren op een dergelijke belofte van iemand anders :).

Beleidsmatig mag de site na release ook geverifieerd worden op toegankelijkheid, als hierover maar verantwoording wordt afgelegd in de toegankelijkheidsverklaring en hier wel concrete stappen voor zijn gezet. Ik vind de ruimte die het huidige toegankelijkheidsbeleid ons bied echter niet zo interessant, alles moet toegankelijk zijn. De terughoudendheid in mijn uitspraak "zo toegankelijk mogelijk" komt voort uit de korte tijd die ik heb gehad om mee te kijken met de toegankelijkheid van de site. Ik zit in het team voor de Corona-app en daar ligt mijn focus. De ondersteuning die ik deze week dus aan de site heb kunnen geven is best-effort en ik heb niet de tijd een volledig onderzoek op WCAG uit te voeren volgens WCAG-EM, wat het beleid uiteindelijk vraagt om te stellen dat iets aantoonbaar toegankelijk is.

Zonder degelijke audit zeggen dat iets gegarandeerd volledig aan WCAG voldoet zie ik als foutieve informatie en daarom de nuance in mijn stelling dat we "zo toegankelijk mogelijk" zijn. In dit geval betekent dat dus dat we er wel degelijk naar gekeken hebben en moeite voor hebben gedaan, maar dar ik de volledige WCAG-compliance nu niet kan garanderen. Ik hoop dat dit wat meer context geeft. Als je concrete toegankelijkheidsissues ziet die nog niet in dit issue benoemd zijn, open dan een nieuw issue en mention mij.

**Comment by mellewynia** [comment]
Stand van zaken, algemeen: een (formele) audit dient natuurlijk nog uitgevoerd te worden, maar er is al een hands-on sessie gedaan met @bramd en Iacobien van Firm Ground. Ook heeft de community aandachtspunten aangedragen en leden van het team van VWS opmerkingen gemaakt. Het meeste is opgelost, maar er zijn nog wat restpunten bekend op dit moment. Waaronder bij mij nu bekend:

- Toevoegen pauzeer-knop bij carroussel (slider).
    - Dan hebben we ook het probleem verholpen dat de iPhone home-button (de visual) er klikbaar uitziet.
- Nalopen van contrast van carroussel bullets (stippen)
- Technisch: Footer links: Cookies, Privacy moet ul-li-HTML structuur volgen.
- Technisch: Powermapper (en Axe) controle nader nalopen, in principe voldoende (o.b.v. hands-on sessie).
- Technisch: toegankelijkheid SVG-afbeeldingen nalopen (leek goed bij smoketest).

Verder hebben we andere UX / UI bevindingen:
- Veelgestelde vragen: vragen worden verborgen / getoond en dit (b)lijkt niet intuïtief.

Opmerking: woensdag 8 juli is de website door Job Jansweijer & Co. een usability test gedaan met een 60+ doelgroep. Hieruit zijn ook bevindingen te verwachten.

**Comment by mellewynia** [comment]
Zie #91 voor voortzetting (post-launch).
